File: db4o.net/Db4objects.Db4o.Tests/Db4objects.Db4o.Tests/Common/Fieldindex/ComplexFieldIndexItem.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using Db4objects.Db4o.Tests.Common.Fieldindex;

namespace Db4objects.Db4o.Tests.Common.Fieldindex
{
	public class ComplexFieldIndexItem : IHasFoo
	{
		public int foo;

		public int bar;

		public Db4objects.Db4o.Tests.Common.Fieldindex.ComplexFieldIndexItem child;

		public ComplexFieldIndexItem()
		{
		}

		public ComplexFieldIndexItem(int foo_, int bar_, Db4objects.Db4o.Tests.Common.Fieldindex.ComplexFieldIndexItem
			 child_)
		{
			foo = foo_;
			bar = bar_;
			child = child_;
		}

		public virtual int GetFoo()
		{
			return foo;
		}
	}
}

File: db4oj.tests/src/com/db4o/db4ounit/common/fieldindex/ComplexFieldIndexItem.java
/* Copyright (C) 2007   Versant Inc.   http://www.db4o.com */
package com.db4o.db4ounit.common.fieldindex;

public class ComplexFieldIndexItem implements HasFoo {
	public int foo;
	public int bar;
	public ComplexFieldIndexItem child;
	
	public ComplexFieldIndexItem() {
	}
	
	public ComplexFieldIndexItem(int foo_, int bar_, ComplexFieldIndexItem child_) {
		foo = foo_;
		bar = bar_;
		child = child_;
	}
	
	public int getFoo() {
		return foo;
	}
}
